### PR TITLE
redefine "pegout" to better match elementsd RPC

### PR DESCRIPTION
**Pull request by apoelstra**

### CODE REVIEW

**Comment by stevenroose** [comment]
LGTM. I built hal-elements with this.
`hal elements tx decode` for the second test vector tx panic'ed with `Execution failed: internal error: entered unreachable code` while with this PR, it shows a pegout-free transaction response.

**Comment by apoelstra** [comment]
Rebased; changed version to 0.7.1 rather than 0.6.1

**Comment by roconnor-blockstream** [comment]
LGTM

**Comment by instagibbs** [comment]
Appears to sync with elementsd. utACK

**Comment by roconnor-blockstream** [comment]
See also https://github.com/ElementsProject/elements/issues/598#issuecomment-487125510

**Comment by apoelstra** [comment]
Rebased.

**Comment by apoelstra** [comment]
Looks good I believe with https://github.com/ElementsProject/elements/pull/610 both rust-elements and elements should agree.

File: src/transaction.rs [hunk 1]
```diff
@@ -400,23 +404,37 @@ impl TxOut {
 
         // Parse destination scriptpubkey
         let script_pubkey = if let Some(Instruction::PushBytes(data)) = iter.next() {
-            Script::from(data.to_owned())
+            if data.is_empty() {
```
Comment (roconnor-blockstream) [inline src/transaction.rs]: Elements was modified to make 0 byte pushes legal @ https://github.com/ElementsProject/elements/pull/610/files#diff-f7ca24fb80ddba0f291cb66344ca6fcbR222

So this block needs to be reverted now.

Comment (instagibbs) [reply src/transaction.rs]: not merged yet! but looks like ACKs from relevant parties

Comment (roconnor-blockstream) [reply src/transaction.rs]: oh indeed.

Comment (roconnor-blockstream) [reply src/transaction.rs]: 610 is now merged and this needs to be reverted now.

Comment (apoelstra) [reply src/transaction.rs]: https://github.com/ElementsProject/rust-elements/issues/20

**Pull request merged**